packages/blocks/src/lib/navigation-map/components/input/geometry.svelte

```diff
@@ -0,0 +1,96 @@
+<script lang='ts'>
+
+import { Radio } from '@viamrobotics/prime-core'
+import { createEventDispatcher } from 'svelte';
+import type { Geometry, Shapes } from '$lib';
+import VectorInput from '../vector-input.svelte';
+import { createGeometry } from '../../lib/geometry';
+
+export let geometry: Geometry;
+
+const dispatch = createEventDispatcher<{ input: Geometry }>();
+
+const handleShapeSelect = (event: CustomEvent<{ value: string }>) => {
+  const nextType = event.detail.value.toLowerCase() as Shapes;
+  dispatch('input', createGeometry(nextType));
+};
+
+const handleDimensionsInput = (event: CustomEvent<Record<string, number>>) => {
+  switch (geometry.type) {
+    case 'box': {
+      const length = event.detail['Length (m)'];
+      const width = event.detail['Width (m)'];
+      const height = event.detail['Height (m)'];
+
+      if (length !== undefined) {
+        geometry.length = length;
```

Comment:
Re-commenting due to diff so this problem doesn't get lost: this function modifies the `geometry` object passed in as a prop, which _will_ change its value in the parent component, which is not good

We should probably start marking prop objects as `Readonly`

Reply:
In this case I'm intentionally mutation because this block will run alongside the frameloop whenever users are resizing geometry - I think I might have a solution that I like much better than this though that I think you'll be more comfortable with, stay tuned!